functions/util/authGuard.js

```diff
@@ -0,0 +1,34 @@
+const { admin, db } = require('./admin')
+const httpStatus = require('http-status-codes')
+const { status, message } = require('./constants')
+
+module.exports = async (req, res, next) => {
+    const { UNAUTHORIZED } = httpStatus
+    const { error, fail } = status
+    const { unAuthorized, somethingWentWrong } = message
+
+    let idToken
```

Comment:
Why do you like using let here

Reply:
using let outside that scope because i was going to make use it outside

Reply:
But it doesn't look like you are so move it to its correct scope